### DISCUSSION TITLE
Cant switch to any renderer other than D3D11 and D3D11 (Experimental) "The selected renderer failed to initialise"

### DISCUSSION THREAD

**Discussion by Likver** [ppy/osu#29756] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Trying to switch to any renderer thats not D3D11 or D3D11 (Experimental) doesnt work and automatically switches back to default renderer (D3D11). This happens always, no matter the screen mode (fullscreen, windowed, etc), frame limiter setting i use or threading mode, this behavior also doesnt change wether i switch from D3D11 or from D3D11 (Experimental)

Im currently running on win8.1 64bit (might have something to do with the issue)

Please let me know if i forgot to add something to my report and i will add it, im not used to do issue reports so i might have done something wrong or not done fully.

### Screenshots or videos

Trying to switch to D3D11 (Experimental) = it works (visual glitches everywhere but it works, these screenshots are all 1 after the other just sitting on main menu)
![image](https://github.com/user-attachments/assets/d2a1dc4c-990c-4aaf-a4c1-71a7752b7786)
![image](https://github.com/user-attachments/assets/e9af5036-ed8e-49ab-8c1e-7b2ed5a1afac)
![image](https://github.com/user-attachments/assets/232b2d23-8bc4-4909-adea-cfc7eb1b98e0)

Trying to switch to D3D11 = it works as expected
![image](https://github.com/user-attachments/assets/38bdaa35-945d-4052-b819-d3a5c6865a1f)

Trying to switch to OpenGL = it doesnt switch and switches back to default which is D3D11
![image](https://github.com/user-attachments/assets/af02615d-0e01-4bdd-b0eb-ccc58e578672)

Trying to switch to Vulkan (Experimental) = it doesnt switch and switches back to default which is D3D11
![image](https://github.com/user-attachments/assets/9d967b77-7a54-40d9-97b6-99abc068fa06)

### Version

2024.906.1

### Logs

idk if it could make some difference but i can put the logs switching from D3D11 to the other renderers, from D3D11 (Experimental) to the other renderers, and from D3D11 and D3D11 (Experimental) after trying to do the renderer switch. 

Since im not sure it could make any difference on the logs, im gonna wait and if u need me to, i can send those logs after switching from every renderer to every renderer.

This log was exported while on D3D11 if it is useful to know
[compressed-logs.zip](https://github.com/user-attachments/files/16914208/compressed-logs.zip)

**Comment by Joehuu** [comment]
I assume your CPU is first-gen intel based on [this wikipedia article](https://en.wikipedia.org/wiki/Intel_Graphics_Technology#Capabilities_(GPU_hardware))? Direct3D feature level matches, lazer needs at least OpenGL 3, and Vulkan is not supported on your integrated graphics.

Converting to discussion because of old hardware.

**Reply by Likver** [reply]
My cpu is the [Celeron B815](https://www.intel.com/content/www/us/en/products/sku/63919/intel-celeron-processor-b815-2m-cache-1-60-ghz/specifications.html), which is based on 2nd gen, and based on the same article u linked, it supports up to openGL 3.1. 
![thorium_40jF196cFl](https://github.com/user-attachments/assets/d2e77ac5-5795-4d7e-a5b9-3b76d22a497f)

Although i should have expected vulkan not to work (on windows at least, because on an emulator i was able to choose and use vulkan on linux mint without any issue), i did expect openGL to work (on windows) for that reason.

I also found [this](https://www.intel.com/content/www/us/en/support/articles/000005524/graphics.html) on the official intel page
![thorium_n8KhxGrpvJ](https://github.com/user-attachments/assets/9e988894-47b8-4409-809b-b8024ca03c4b)

**Reply by sineplusx** [reply, marked as answer]
I think OpenGL 3.2 is the current minimum version required for lazer, as documented in #22966
